src/test/ui/async-await/in-trait/async-example-desugared-in-trait.rs

```diff
@@ -0,0 +1,20 @@
+// check-pass
+// edition: 2021
+
+#![feature(async_fn_in_trait)]
+#![feature(return_position_impl_trait_in_trait)]
+#![allow(incomplete_features)]
+
+use std::future::Future;
+
+trait MyTrait {
+    fn foo(&self) -> impl Future<Output = i32> + '_;
+}
+
+impl MyTrait for i32 {
+    async fn foo(&self) -> i32 {
```

Comment:
Similarly, this can't be marked with `// known-bug`, but this will be rejected by a PR that implements #102745, so perhaps say that this should/will be broken.